docs/changelog.rst

```diff
@@ -1,8 +1,19 @@
 :orphan:
 
+.. See https://releases.readthedocs.io/ for instructions
+
 =========
 Changelog
 =========
+
+* :feature:`231` Add PEP 561 compatible type annotations using mypy
+* :feature:`602` Add stricter validation of repository URL
+* :bug:`605` Fix inaccurate retry message
+* :bug:`598` Fix vague error message for archive format
+* :support:`634` Add linting for docstrings
+* :support:`-` Update and expand documentation
+* :support:`-` Add test coverage
+* :support:`-` Format code with black and isort
```

Comment:
I attempted to be explicit re: user-facing changes, while condensing the `support` changes. I'm happy to tweak/rewrite this.

Reply:
The changelog should be about *user-facing* changes. I don't know what `support` means but no user wants the cognitive overhead of reading a changelog with things that don't apply to them.

The rest of these changes need to describe how they help the user. For example, "Add PEP 561 compatible type annotations and package them for users of Twine's API". What does stricter validation of the repository URL mean? How do users fix their URLs to accommodate it?

Reply:
>  I don't know what `support` means

From https://releases.readthedocs.io/en/latest/concepts.html: "Support items vary in size but are usually non-code-related changes, such as documentation or packaging updates."

> The changelog should be about _user-facing_ changes.

Even after reading the existing changelog, this wasn't clear to me. Personally, I see value in showing high-level contributor-facing changes here as well, e.g. to indicate the overall health of the project. I've seen those sorts of entries in other projects

It seems like there's some prior art for both these [starting at 1.10.0](https://twine.readthedocs.io/en/latest/changelog.html#1.10.0).

But, I don't feel strongly. Leaving them out makes maintaining the changelog easier.

Reply:
I made some tweaks in https://github.com/pypa/twine/pull/643/commits/7ed25773597bc5a47c8e001222cb7e141c0c4ebd.

One was to link to the PR's, instead of the issues, because those seemed to have more useful information about what changed, along with a clear link back to the original issue.

I didn't see a clear pattern from previous changelog entries.

Reply:
> Personally, I see value in showing high-level contributor-facing changes here as well, e.g. to indicate the overall health of the project.

There are better ways to display that.

> It seems like there's some prior art for both these [starting at 1.10.0](https://twine.readthedocs.io/en/latest/changelog.html#1.10.0).

1.10 has items in there that affect the end user (proper rendering, organization, etc.) There are a handful of releases after that that _do_ have information not important to users. Because we've made that mistake before, doesn't make it justified to continue to do it.

Users looking to upgrade shouldn't have to try to figure out what's important to them.

Reply:
Removed entries in https://github.com/pypa/twine/pull/643/commits/d1b8c1f786c23aea5207ab5a4d40396659e0f346 (aside: is this kind of comment redundant/irritating?)

> > Personally, I see value in showing high-level contributor-facing changes here as well, e.g. to indicate the overall health of the project.
> 
> There are better ways to display that.

What do you have in mind?

Reply:
There are ways to document people's contributions without littering the changelog. We're discussing this in a few places now (AUTHORS and similar files is one way to do it. See how urllib3 does this by having a list of the things folks have done)

Reply:
To be clear, @bhrutledge I don't think you're littering here. I realize now how this came across. I'm sure you've heard the "signal to noise ratio" argument before and that's exactly how I think of Changelogs. They need to be high signal, and low noise. That's why I'm so laser focused on them being very user-centric and using other forms of documentation for acknowledging and expressing appreciation to contributors. It's of course hard to express that quickly, as I'm trying to move through GitHub notifications quickly in the morning in the few minutes I have before work. I'll try to wait to respond until I have more time